### PR TITLE
[stdlib] Obsolete various compatibility shims in 5.0

### PR DESCRIPTION
**Pull request by airspeedswift**
Taking another run at this...

It's a lot of changes but mainly because we template out a zillion variation tests to eliminate modifiers on Slice from the templated tests..

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please test

**Comment by airspeedswift** [comment]
@swift-ci please test source compatibility

**Comment by airspeedswift** [comment]
@swift-ci please test source compatibility

**Comment by airspeedswift** [comment]
@swift-ci please smoke test

File: validation-test/stdlib/Lazy.swift.gyb [hunk 1]
```diff
@@ -124,21 +124,20 @@ LazyTestSuite.test("CollectionOfOne/AssociatedTypes") {
...
-%   ('subscript(Index)/Set', 'c[i] = OpaqueValue(42)', '-2, -1, 1, 2'),
-% ]:
-LazyTestSuite.test("CollectionOfOne/${name}")
-  .forEach(in: [${indices}]) {
-  i in
-
-  var c = CollectionOfOne<OpaqueValue<Int>>(OpaqueValue(42))
-  expectCrashLater()
-  ${operation}
+let collectionOfOneIndexTests: [(String,(inout CollectionOfOne<OpaqueValue<Int>>,Int)->Void,[Int])] = [
```
Comment (moiseev) [inline validation-test/stdlib/Lazy.swift.gyb]: :+1:

File: validation-test/stdlib/Lazy.swift.gyb [hunk 2]
```diff
@@ -241,27 +240,26 @@ LazyTestSuite.test("EmptyCollection/AssociatedTypes") {
...
-%   ('subscript(Index)/Set', 'c[i] = OpaqueValue(42)'),
-% ]:
-LazyTestSuite.test("EmptyCollection/${name}")
-  .forEach(in: [-1, 0, 1]) {
-  i in
-
-  var c = EmptyCollection<OpaqueValue<Int>>()
-  expectCrashLater()
-  ${operation}
+let emptyCollectionIndexTests: [(String,(inout EmptyCollection<OpaqueValue<Int>>,Int)->Void)] = [
```
Comment (moiseev) [inline validation-test/stdlib/Lazy.swift.gyb]: s/Void/() would make the line shorter =)

**Review by moiseev — APPROVED** [review]
LGTM

**Comment by moiseev** [comment]
Should we squash the tiny commits?

File: validation-test/stdlib/Lazy.swift.gyb (hunk 2 above)
Comment (airspeedswift) [reply validation-test/stdlib/Lazy.swift.gyb]: I think `()` for arguments, `Void` for return is our style tho.

**Comment by airspeedswift** [comment]
I'll just squash the whole thing on merge.

**Pull request merged**